packages/chrome/src/elements/sheet/Sheet.tsx

```diff
@@ -67,16 +67,27 @@ export const Sheet = React.forwardRef<HTMLElement, ISheetProps>(
     const sheetRef = useRef<HTMLElement>(null);
 
     const seed = useUIDSeed();
+    const [isCloseButtonPresent, setCloseButtonPresent] = useState<boolean>(false);
     const [idPrefix] = useState<string>(id || seed(`sheet_${PACKAGE_VERSION}`));
     const titleId = `${idPrefix}--title`;
     const descriptionId = `${idPrefix}--description`;
 
-    const sheetContext = useMemo(() => ({ titleId, descriptionId }), [titleId, descriptionId]);
+    const sheetContext = useMemo(
+      () => ({
+        titleId,
+        descriptionId,
+        isCloseButtonPresent,
+        setCloseButtonPresent(isPresent: boolean) {
+          setCloseButtonPresent(isPresent);
+        }
+      }),
+      [titleId, descriptionId, isCloseButtonPresent]
+    );
 
     useFocusableMount({ targetRef: sheetRef, isMounted: isOpen, focusOnMount, restoreFocus });
 
     return (
-      <SheetContext.Provider value={sheetContext}>
+      <SheetContext.Provider value={sheetContext as ISheetContext}>
```

Comment:
is there a means for typing `useMemo` to indicate the return type and avoid `as`?

Reply:
Ahh good call, let me take a go at that.